app/models/lookup_key.rb

```diff
@@ -214,7 +207,7 @@ def array2path(array)
   def validate_and_cast_default_value
     return true if default_value.nil? || contains_erb?(default_value)
     begin
-      self.default_value = cast_validate_value self.default_value
+      Foreman::Parameters::Caster.new(self, :attribute_name => :default_value, :to => key_type.to_s.to_sym).cast!
```

Comment:
It can be nil, in which case, to sym will fail.
On Aug 11, 2015 12:12 PM, "Dominic Cleal" <notifications@github.com> wrote:

> In app/models/lookup_key.rb
> <https://github.com/theforeman/foreman/pull/2399#discussion_r36725487>:
>
> > @@ -214,7 +207,7 @@ def array2path(array)
> >    def validate_and_cast_default_value
> >      return true if default_value.nil? || contains_erb?(default_value)
> >      begin
> > -      self.default_value = cast_validate_value self.default_value
> > +      Foreman::Parameters::Caster.new(self, :attribute_name => :default_value, :to => key_type.to_s.to_sym).cast!
>
> .to_s is redundant here I think, since key_type's a string column
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theforeman/foreman/pull/2399/files#r36725487>.
>